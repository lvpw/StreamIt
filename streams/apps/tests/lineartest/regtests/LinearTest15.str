/** Test out combining splitjoins. **/
void->void pipeline LinearTest15 {
  add FloatSource();
  add LinearSJ();
  add FloatSink();
}

void->float stateful filter FloatSource {
  float x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = x + 1;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

/** A linear split join to test combination with different round robin weightings. **/
float->float splitjoin LinearSJ {
  split roundrobin(2,3);
  add FloatFilter1();
  add FloatFilter2();
  join roundrobin;
}


/** filter that computes [[1 2][3 4]] **/
float->float filter FloatFilter1 {
  work pop 2 push 2 {
    float t1 = pop();
    float t2 = pop();
    push(t2*2 + t1*4);
    push(t2*1 + t1*3);
  }
}
/** filter that computes [[5 6][7 8][9 10]] **/
float->float filter FloatFilter2 {
  work pop 3 push 2 {
    float t1 = pop();
    float t2 = pop();
    float t3 = pop();
    push(10*t1 + 8*t2 + 6*t3);
    push(9*t1  + 7*t2 + 5*t3);
  }
}
