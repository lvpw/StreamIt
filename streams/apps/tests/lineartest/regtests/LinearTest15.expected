FloatFilter1_1.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]]
FloatFilter1_1.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter2_1.matrix
[[5.0+0.0i 6.0+0.0i]
 [7.0+0.0i 8.0+0.0i]
 [9.0+0.0i 10.0+0.0i]]
FloatFilter2_1.vector
[[0.0+0.0i 0.0+0.0i]]
LinearSJ_1.matrix
[[5.0+0.0i 0.0+0.0i 6.0+0.0i 0.0+0.0i]
 [7.0+0.0i 0.0+0.0i 8.0+0.0i 0.0+0.0i]
 [9.0+0.0i 0.0+0.0i 10.0+0.0i 0.0+0.0i]
 [0.0+0.0i 1.0+0.0i 0.0+0.0i 2.0+0.0i]
 [0.0+0.0i 3.0+0.0i 0.0+0.0i 4.0+0.0i]]
LinearSJ_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
